### PR TITLE
Sonar fixes

### PR DESCRIPTION
**Pull request by ham1**
## Description
* Removed `clone()` methods which just called `super.clone()`
* Reduced size/complexity of methods by breaking them into smaller ones and/or reducing nesting
* Replaced `size() == 0` with `isEmpty()`
* Formatting
* Removed blank or useless comments

## How Has This Been Tested?
ant test

## Types of changes
- Dev fix

## Checklist:

- [x] My code follows the [code style][style-guide] of this project.
- [x] I have updated the documentation accordingly.

[style-guide]: https://wiki.apache.org/jmeter/CodeStyleGuidelines

### CODE REVIEW

**Comment by pmouawad** [comment]
Hi @ham1 ,
Thanks for this PR.
Unfortunately I am not very comfortable with PR that touch a lot of file and where the code modification may touch more things than what the PR describes.

Would it be possible to rebase your PR and split it into at max 10 files ?
It is easier to review and merging is faster for me, which will mean I'll merge it earlier and you won't have to rebase.

Also if possible, try to make the PR really only touch what it pretends to.
Sonar fixes might sometimes break existing code so they need careful review and merge.
Thanks a lot again for all your work !

Regards

**Comment by ham1** [comment]
I've split up the PR into multiple commits, hopefully these are easier to merge; please let me know if not.

File: src/components/org/apache/jmeter/assertions/BSFAssertion.java [hunk 1]
```diff
@@ -57,8 +57,4 @@ public AssertionResult getResult(SampleResult response) {
         return result;
     }
     
-    @Override
-    public Object clone() {
```
Comment (FSchumacher) [inline src/components/org/apache/jmeter/assertions/BSFAssertion.java]: If we remove this method sonar will complain about the missing method, as the class is marked Clonable

File: src/components/org/apache/jmeter/assertions/HTMLAssertion.java [hunk 2]
```diff
@@ -328,17 +304,13 @@ public void setXML() {
...
      */
     public boolean isXML() {
         return getPropertyAsLong(FORMAT_KEY) == 2;
     }
 
     /**
-     * Sets the name of the file where tidy writes the output to
-     * 
-     * @return name of file
+     * Gets the name of the file where tidy writes the output to
```
Comment (FSchumacher) [inline src/components/org/apache/jmeter/assertions/HTMLAssertion.java]: If we remove the return tag, javadoc will complain.

**Comment by pmouawad** [comment]
Hello @ham1 ,
I hope you're fine.
Thanks for this PR which took some time to be taken into account. 

It now conflicts with a lot of files so it would be nice if you could rebase it.

Anyway, I suggest for easier& faster integration, that similar PRs never touch more than 10 files.
I'll be happy to merge them very fastly. But whenever it exceeds such number it becomes harder to merge rapidly and the conflict risk is higher.

Thank you

**Comment by ham1** [comment]
Sorry, don't have enough time to complete this, so I will close it for now.
Thanks for merging some of the changes and I'll try to submit smaller ones in future :smile:

**Pull request closed without merging**